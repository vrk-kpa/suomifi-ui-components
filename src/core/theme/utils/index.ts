export {
  withSuomifiDefaultProps,
  themeOrTokens,
  TokensOrThemeProps,
} from './defaults';
export { focus } from './focus';
export { boxshadowOutline } from './outline';
export { margin, padding, spacingModifiers } from './spacing';
export { typographyUtils, fontSize } from './typography';
