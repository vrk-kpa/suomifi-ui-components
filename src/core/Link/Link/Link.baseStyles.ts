import { SuomifiTheme } from '../../theme';
import { css } from 'styled-components';
import { baseStyles } from '../BaseLink/BaseLink.baseStyles';
import { MarginProps, buildSpacingCSS } from '../../theme/utils/spacing';

export const LinkStyles = (
  theme: SuomifiTheme,
  globalMargins?: MarginProps,
  propMargins?: MarginProps,
) => css`
  ${buildSpacingCSS(globalMargins)}
  ${buildSpacingCSS(propMargins, true)}
  ${baseStyles(theme)}
`;
