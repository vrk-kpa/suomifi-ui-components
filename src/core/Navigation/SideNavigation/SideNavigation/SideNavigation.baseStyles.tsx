import { font } from '../../../theme/reset';
import { css } from 'styled-components';
import { SuomifiTheme } from '../../../theme';
import { MarginProps, buildSpacingCSS } from '../../../theme/utils/spacing';

export const baseStyles = (
  theme: SuomifiTheme,
  globalMargins?: MarginProps,
  propMargins?: MarginProps,
) => css`
  ${font(theme)('bodyText')}
  ${buildSpacingCSS(globalMargins)}
  ${buildSpacingCSS(propMargins, true)}
  &.fi-side-navigation {
    .fi-side-navigation_divider {
      height: 1px;
      background: ${theme.colors.depthSecondary};
      margin: ${theme.spacing.s};
    }

    .fi-side-navigation_heading {
      ${font(theme)('heading4')}
      display: flex;
      justify-content: space-between;
      align-items: center;
      padding: ${theme.spacing.xs} ${theme.spacing.s} 0 ${theme.spacing.s};

      .fi-side-navigation_heading_inner {
        display: flex;
        align-items: center;
      }

      .fi-icon {
        height: 45px;
        width: 45px;
        margin-right: ${theme.spacing.s};
      }
    }

    .fi-side-navigation_list {
      list-style-type: none;
      margin: 0;
      padding: 0;
    }

    &--small-screen {
      border: 1px solid ${theme.colors.depthLight1};
      nav {
        padding: 0 ${theme.spacing.xxs};
      }
      .fi-side-navigation_divider {
        margin: ${theme.spacing.xxs} ${theme.spacing.s} 0 ${theme.spacing.s};
      }

      .fi-side-navigation_heading {
        ${font(theme)('heading4')}
        width: 100%;
        padding: ${theme.spacing.s};

        &:focus {
          position: relative;
          outline: 0;

          &:after {
            ${theme.focuses.absoluteFocus}
          }
        }

        .fi-icon {
          height: 25px;
          width: 25px;
          color: ${theme.colors.highlightBase};
        }
      }
    }
  }
`;
