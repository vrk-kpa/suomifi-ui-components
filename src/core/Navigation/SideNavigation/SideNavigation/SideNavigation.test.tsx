import React from 'react';
import { render } from '@testing-library/react';
import { axeTest } from '../../../../utils/test';

import { SideNavigation } from './SideNavigation';
import { SideNavigationItem } from '../SideNavigationItem/SideNavigationItem';
import { ExternalLink, RouterLink } from '../../../Link';
import { IconPiggyBank } from 'suomifi-icons';

interface TestProps {
  children: string;
  onClick?: () => void;
  role?: string;
  tabIndex?: number;
}

const CustomButton = (props: TestProps) => {
  const { children, ...passProps } = props;
  return <button {...passProps}>{props.children}</button>;
};

const TestSideNavigation = (
  <SideNavigation heading="Economy" icon={<IconPiggyBank />} aria-label="Test">
    <SideNavigationItem
      subLevel={1}
      content={
        <RouterLink href="/" aria-current="location">
          Personal economy
        </RouterLink>
      }
    >
      <SideNavigationItem
        subLevel={2}
        content={
          <RouterLink href="/" aria-current="location">
            Crisis situations in personal finances
          </RouterLink>
        }
      >
        <SideNavigationItem
          subLevel={3}
          content={
            <RouterLink href="/">
              If you are unable to pay your debts
            </RouterLink>
          }
        />
        <SideNavigationItem
          subLevel={3}
          selected
          content={
            <RouterLink href="/" aria-current="page">
              Advice on banking and insurance matters
            </RouterLink>
          }
        />
      </SideNavigationItem>
      <SideNavigationItem
        subLevel={2}
        content={<RouterLink href="/">Last will and testament</RouterLink>}
      />
    </SideNavigationItem>
    <SideNavigationItem
      subLevel={1}
      content={
        <RouterLink asComponent={ExternalLink} href="https://suomi.fi" hideIcon>
          Taxation and public economy
        </RouterLink>
      }
    />
    <SideNavigationItem
      content={
        <RouterLink
          asComponent={CustomButton}
          onClick={() => console.log('Nav item clicked')}
        >
          Consumer protection
        </RouterLink>
      }
      subLevel={1}
    />
    <SideNavigationItem
      disabled
      content={<RouterLink>Finance</RouterLink>}
      subLevel={1}
    />
  </SideNavigation>
);

test('calling render with the same component on the same container does not remount', () => {
  const NavRendered = render(TestSideNavigation);
  const { container } = NavRendered;
  expect(container.firstChild).toMatchSnapshot();
});

test('should not have basic accessibility issues', axeTest(TestSideNavigation));

describe('margin', () => {
  it('should have margin style from margin prop', () => {
    const { container } = render(
      <SideNavigation heading="" aria-label="" margin="xs">
        <SideNavigationItem content="" subLevel={1} />
      </SideNavigation>,
    );
    expect(container.firstChild).toHaveStyle('margin: 10px');
  });

  it('should have margin style overridden by style prop', async () => {
    const { container } = render(
      <SideNavigation
        heading=""
        aria-label=""
        margin="xs"
        style={{ margin: 2 }}
      >
        <SideNavigationItem content="" subLevel={1} />
      </SideNavigation>,
    );
    expect(container.firstChild).toHaveAttribute('style', 'margin: 2px;');
  });
});
