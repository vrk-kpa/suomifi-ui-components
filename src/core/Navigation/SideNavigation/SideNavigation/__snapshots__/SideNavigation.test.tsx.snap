// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calling render with the same component on the same container does not remount 1`] = `
.c8 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  background-color: transparent;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c8::before,
.c8::after {
  box-sizing: border-box;
}

.c8:hover,
.c8:active,
.c8:focus,
.c8:focus-within {
  color: inherit;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
}

.c0 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c0::before,
.c0::after {
  box-sizing: border-box;
}

.c6 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: list-item;
}

.c6::before,
.c6::after {
  box-sizing: border-box;
}

.c4 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
}

.c4::before,
.c4::after {
  box-sizing: border-box;
}

.c2 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c2::before,
.c2::after {
  box-sizing: border-box;
}

.c5 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  list-style-type: decimal;
  margin-top: 1em;
  margin-bottom: 1em;
  margin-left: 0;
  margin-right: 0;
  padding-left: 40px;
}

.c5::before,
.c5::after {
  box-sizing: border-box;
}

.c10 {
  vertical-align: baseline;
}

.c10.fi-icon {
  display: inline-block;
}

.c10 .fi-icon-base-fill {
  fill: currentColor;
}

.c10 .fi-icon-base-stroke {
  stroke: currentColor;
}

.c10.fi-icon--cursor-pointer {
  cursor: pointer;
}

.c10.fi-icon--cursor-pointer * {
  cursor: inherit;
}

.c3 {
  vertical-align: baseline;
}

.c3.fi-static-icon {
  display: inline-block;
}

.c3.fi-icon--cursor-pointer {
  cursor: pointer;
}

.c3.fi-icon--cursor-pointer * {
  cursor: inherit;
}

.c1 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
}

.c1.fi-side-navigation .fi-side-navigation_divider {
  height: 1px;
  background: hsl(215,100%,97%);
  margin: 15px;
}

.c1.fi-side-navigation .fi-side-navigation_heading {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 24px;
  font-weight: 600;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 15px 0 15px;
}

.c1.fi-side-navigation .fi-side-navigation_heading .fi-side-navigation_heading_inner {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1.fi-side-navigation .fi-side-navigation_heading .fi-icon {
  height: 45px;
  width: 45px;
  margin-right: 15px;
}

.c1.fi-side-navigation .fi-side-navigation_list {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

.c1.fi-side-navigation--small-screen {
  border: 1px solid hsl(202,7%,80%);
}

.c1.fi-side-navigation--small-screen nav {
  padding: 0 5px;
}

.c1.fi-side-navigation--small-screen .fi-side-navigation_divider {
  margin: 5px 15px 0 15px;
}

.c1.fi-side-navigation--small-screen .fi-side-navigation_heading {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 24px;
  font-weight: 600;
  width: 100%;
  padding: 15px;
}

.c1.fi-side-navigation--small-screen .fi-side-navigation_heading:focus {
  position: relative;
  outline: 0;
}

.c1.fi-side-navigation--small-screen .fi-side-navigation_heading:focus:after {
  content: '';
  position: absolute;
  pointer-events: none;
  top: -2px;
  right: -2px;
  bottom: -2px;
  left: -2px;
  border-radius: 2px;
  background-color: transparent;
  border: 0px solid hsl(0,0%,100%);
  box-sizing: border-box;
  box-shadow: 0 0 0 2px hsl(196,77%,44%);
  z-index: 9999;
}

.c1.fi-side-navigation--small-screen .fi-side-navigation_heading .fi-icon {
  height: 25px;
  width: 25px;
  color: hsl(212,63%,45%);
}

.c7 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 600;
}

.c7.fi-side-navigation-item {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  margin: 5px 0;
}

.c7.fi-side-navigation-item:focus,
.c7.fi-side-navigation-item:focus-within {
  outline: none;
  box-shadow: none;
  border: none;
}

.c7.fi-side-navigation-item .fi-link--router {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 10px 10px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  cursor: pointer;
  border: none;
  color: hsl(212,63%,45%);
  background: transparent;
}

.c7.fi-side-navigation-item .fi-link--router:visited {
  color: hsl(212,63%,45%);
}

.c7.fi-side-navigation-item .fi-link--router:hover,
.c7.fi-side-navigation-item .fi-link--router:active {
  background: hsl(212,63%,45%);
  -webkit-text-decoration: none;
  text-decoration: none;
  color: hsl(0,0%,100%);
  border-radius: 4px;
}

.c7.fi-side-navigation-item .fi-link--router:focus-visible {
  outline: 3px solid transparent;
  position: relative;
  box-shadow: none;
}

.c7.fi-side-navigation-item .fi-link--router:focus-visible:after {
  content: '';
  position: absolute;
  pointer-events: none;
  top: -2px;
  right: -2px;
  bottom: -2px;
  left: -2px;
  border-radius: 2px;
  background-color: transparent;
  border: 0px solid hsl(0,0%,100%);
  box-sizing: border-box;
  box-shadow: 0 0 0 2px hsl(196,77%,44%);
  z-index: 9999;
}

.c7.fi-side-navigation-item.fi-side-navigation-item--child-selected > span > .fi-link--router {
  font-weight: 600;
}

.c7.fi-side-navigation-item .fi-side-navigation-item_sub-list {
  list-style-type: none;
  padding: 0;
  margin: 0;
}

.c7.fi-side-navigation-item .fi-side-navigation-item_content-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.c7.fi-side-navigation-item .fi-side-navigation-item_content-wrapper .fi-icon {
  width: 10px;
  height: 10px;
  pointer-events: none;
  color: hsl(212,63%,45%);
  padding: 0 10px;
}

.c7.fi-side-navigation-item--level-1.fi-side-navigation-item--selected > span > .fi-link--router {
  color: hsl(0,0%,100%);
  font-weight: 600;
  background: hsl(212,63%,45%);
  border-radius: 4px;
}

.c7.fi-side-navigation-item--level-2.fi-side-navigation-item--selected > span > .fi-link--router {
  font-weight: 600;
  background: hsl(212,63%,45%);
  color: hsl(0,0%,100%);
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
}

.c7.fi-side-navigation-item--level-2.fi-side-navigation-item--selected > span > .fi-link--router + .fi-icon {
  color: hsl(0,0%,100%);
}

.c7.fi-side-navigation-item--level-2 > span > .fi-link--router {
  text-transform: none;
  padding-right: 60px;
  color: hsl(212,63%,45%);
  background: hsl(0,0%,100%);
}

.c7.fi-side-navigation-item--level-2 .fi-side-navigation-item_sub-list {
  margin: 5px 0 5px 30px;
  background: hsl(212,63%,95%);
  padding: 5px 0;
}

.c7.fi-side-navigation-item--level-3 {
  margin: 0;
}

.c7.fi-side-navigation-item--level-3.fi-side-navigation-item--selected > span > .fi-link--router {
  font-weight: 600;
  background: hsl(212,63%,45%);
  color: hsl(0,0%,100%);
}

.c7.fi-side-navigation-item--level-3 > span > .fi-link--router {
  text-transform: none;
  padding-left: 15px;
  color: hsl(212,63%,45%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
}

.c7.fi-side-navigation-item--level-3 > span > .fi-link--router:hover {
  border-radius: 0;
}

.c7.fi-side-navigation-item--disabled > span > .fi-link--router {
  color: hsl(202,7%,67%);
  cursor: not-allowed;
  background: transparent;
}

.c7.fi-side-navigation-item--disabled > span > .fi-link--router:hover,
.c7.fi-side-navigation-item--disabled > span > .fi-link--router:active,
.c7.fi-side-navigation-item--disabled > span > .fi-link--router:visited {
  background: transparent;
  color: hsl(202,7%,67%);
}

.c12 {
  position: absolute;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  width: 1px;
  margin: -1px;
  padding: 0;
  border: 0;
  overflow: hidden;
}

.c11 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
}

.c11.fi-link {
  color: hsl(212,63%,45%);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c11.fi-link:hover,
.c11.fi-link:active,
.c11.fi-link:focus,
.c11.fi-link:focus-within {
  color: hsl(212,63%,45%);
}

.c11.fi-link:focus,
.c11.fi-link:focus-within {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c11.fi-link:focus {
  border-radius: 2px;
  box-shadow: 0 0 0 2px hsl(196,77%,44%);
  outline: 0;
  outline: 3px solid transparent;
}

.c11.fi-link:hover,
.c11.fi-link:active {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c11.fi-link:visited {
  color: hsl(284,36%,45%);
}

.c11.fi-link--initial-underline {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c11.fi-link--initial-underline:focus,
.c11.fi-link--initial-underline:focus-within {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c11.fi-link--initial-underline:hover,
.c11.fi-link--initial-underline:active {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c11.fi-link.fi-link--accent .fi-link--accent_icon {
  font-size: 16px;
  margin-left: -3px;
  margin-right: 2px;
  -webkit-transform: translateY(0.15em);
  -ms-transform: translateY(0.15em);
  transform: translateY(0.15em);
}

.c11.fi-link.fi-link--small {
  font-size: 16px;
}

.c11 .fi-link_icon {
  margin: 0;
  padding-left: 4px;
  font-size: 16px;
  box-sizing: content-box;
}

.c9 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
}

.c9.fi-link {
  color: hsl(212,63%,45%);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c9.fi-link:hover,
.c9.fi-link:active,
.c9.fi-link:focus,
.c9.fi-link:focus-within {
  color: hsl(212,63%,45%);
}

.c9.fi-link:focus,
.c9.fi-link:focus-within {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c9.fi-link:focus {
  border-radius: 2px;
  box-shadow: 0 0 0 2px hsl(196,77%,44%);
  outline: 0;
  outline: 3px solid transparent;
}

.c9.fi-link:hover,
.c9.fi-link:active {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c9.fi-link:visited {
  color: hsl(284,36%,45%);
}

.c9.fi-link--initial-underline {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c9.fi-link--initial-underline:focus,
.c9.fi-link--initial-underline:focus-within {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c9.fi-link--initial-underline:hover,
.c9.fi-link--initial-underline:active {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c9.fi-link.fi-link--accent .fi-link--accent_icon {
  font-size: 16px;
  margin-left: -3px;
  margin-right: 2px;
  -webkit-transform: translateY(0.15em);
  -ms-transform: translateY(0.15em);
  transform: translateY(0.15em);
}

.c9.fi-link.fi-link--small {
  font-size: 16px;
}

<div
  class="c0 fi-side-navigation c1"
>
  <div
    class="c0 fi-side-navigation_heading"
  >
    <span
      class="c2 fi-side-navigation_heading_inner"
    >
      <svg
        aria-hidden="true"
        class="fi-icon c3"
        focusable="false"
        height="1em"
        viewBox="0 0 80 80"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <path
            class="fi-icon-illustrative-base-fill"
            d="M23.611 62.508c-4.253-2.836-7.23-6.604-8.611-10.896L16.904 51c1.24 3.857 3.944 7.261 7.817 9.844l-1.11 1.664M40.255 68.011c-2.825 0-5.603-.363-8.255-1.08l.522-1.93c4.968 1.342 10.509 1.34 15.472-.001l.522 1.93a31.68 31.68 0 0 1-8.261 1.081M57.07 63.058l-1.11-1.664c5.437-3.625 8.555-8.923 8.555-14.535 0-5.991-3.468-11.511-9.515-15.145L56.03 30c6.664 4.005 10.485 10.15 10.485 16.859 0 6.292-3.442 12.196-9.445 16.199"
            fill="#A5ADB1"
          />
          <path
            class="fi-icon-illustrative-base-fill"
            d="M19 52.943h-8c-1.654 0-3-1.346-3-3v-6c0-1.654 1.346-3 3-3h3.296c1.081-2.761 3.332-5.604 6.704-8.466V22l8.14 5.814c2.585-1.015 4.86-1.871 7.86-1.871v2c-2.723 0-4.783.81-7.393 1.836l-.747.293L23 25.886v7.527l-.361.3c-3.516 2.916-5.827 5.868-6.687 8.537l-.223.693H11c-.552 0-1 .449-1 1v6c0 .551.448 1 1 1h8v2M33 72H23V58h2v12h6v-7h2v9M57 72H47v-9h2v7h6V58h2v14"
            fill="#A5ADB1"
          />
          <path
            class="fi-icon-illustrative-base-fill"
            d="M26 45c-2.757 0-5-2.243-5-5h2c0 1.654 1.346 3 3 3s3-1.346 3-3h2c0 2.757-2.243 5-5 5M65 48v-2c2.757 0 5-2.243 5-5h2c0 3.86-3.141 7-7 7"
            fill="#A5ADB1"
          />
          <path
            class="fi-icon-illustrative-highlight-fill"
            d="m49.601 32.8-1.202-1.6A3.966 3.966 0 0 0 50 28c0-2.206-1.794-4-4-4s-4 1.794-4 4c0 .875.277 1.705.801 2.402l-1.598 1.202A5.948 5.948 0 0 1 40 28c0-3.309 2.691-6 6-6s6 2.691 6 6a5.956 5.956 0 0 1-2.399 4.8M47 14h2V8h-2z"
            fill="#E97025"
          />
          <path
            class="fi-icon-illustrative-base-fill"
            d="M52.4 35.8C48.99 33.243 44.28 32 38 32v-2c6.729 0 11.832 1.374 15.6 4.2l-1.2 1.6"
            fill="#A5ADB1"
          />
          <path
            class="fi-icon-illustrative-highlight-fill"
            d="M43 20h2v-6h-2zM47 18h2v-2h-2zM43 12h2v-2h-2z"
            fill="#E97025"
          />
          <path
            class="fi-icon-illustrative-base-fill"
            d="M39 60h2v-2h-2zM43 60h2v-2h-2zM35 60h2v-2h-2zM8 72h64v-2H8z"
            fill="#A5ADB1"
          />
          <path
            d="M0 0h80v80H0z"
          />
        </g>
      </svg>
      Economy
    </span>
  </div>
  <nav
    aria-label="Test"
    class="c4"
  >
    <div
      class="c0 fi-side-navigation_divider"
    />
    <ul
      class="c5 fi-side-navigation_list"
    >
      <li
        class="c6 c7 fi-side-navigation-item fi-side-navigation-item--level-1 fi-side-navigation-item--child-selected"
      >
        <span
          class="c2 fi-side-navigation-item_content-wrapper"
        >
          <a
            aria-current="location"
            class="c8 fi-link fi-link--router c9"
            href="/"
          >
            Personal economy
          </a>
        </span>
        <ul
          class="c5 fi-side-navigation-item_sub-list"
        >
          <li
            class="c6 c7 fi-side-navigation-item fi-side-navigation-item--level-2 fi-side-navigation-item--child-selected"
          >
            <span
              class="c2 fi-side-navigation-item_content-wrapper"
            >
              <svg
                aria-hidden="true"
                class="fi-icon c10"
                focusable="false"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="fi-icon-base-fill"
                  d="M1 6a1 1 0 0 1 1 1v4h18.585l-3.292-3.293a.999.999 0 1 1 1.414-1.414l4.991 4.992c.032.03.062.063.09.098l-.081-.09a1.005 1.005 0 0 1 .293.674v.066l-.003.052L24 12a1.007 1.007 0 0 1-.302.715l-4.991 4.992a.997.997 0 0 1-1.414 0 .999.999 0 0 1 0-1.414L20.585 13H1a1 1 0 0 1-1-1V7a1 1 0 0 1 1-1Z"
                  fill="#222"
                  fill-rule="evenodd"
                />
              </svg>
              <a
                aria-current="location"
                class="c8 fi-link fi-link--router c9"
                href="/"
              >
                Crisis situations in personal finances
              </a>
            </span>
            <ul
              class="c5 fi-side-navigation-item_sub-list"
            >
              <li
                class="c6 c7 fi-side-navigation-item fi-side-navigation-item--level-3"
              >
                <span
                  class="c2 fi-side-navigation-item_content-wrapper"
                >
                  <a
                    class="c8 fi-link fi-link--router c9"
                    href="/"
                  >
                    If you are unable to pay your debts
                  </a>
                </span>
              </li>
              <li
                class="c6 c7 fi-side-navigation-item fi-side-navigation-item--level-3 fi-side-navigation-item--selected"
              >
                <span
                  class="c2 fi-side-navigation-item_content-wrapper"
                >
                  <a
                    aria-current="page"
                    class="c8 fi-link fi-link--router c9"
                    href="/"
                  >
                    Advice on banking and insurance matters
                  </a>
                </span>
              </li>
            </ul>
          </li>
          <li
            class="c6 c7 fi-side-navigation-item fi-side-navigation-item--level-2"
          >
            <span
              class="c2 fi-side-navigation-item_content-wrapper"
            >
              <svg
                aria-hidden="true"
                class="fi-icon c10"
                focusable="false"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="fi-icon-base-fill"
                  d="M1 6a1 1 0 0 1 1 1v4h18.585l-3.292-3.293a.999.999 0 1 1 1.414-1.414l4.991 4.992c.032.03.062.063.09.098l-.081-.09a1.005 1.005 0 0 1 .293.674v.066l-.003.052L24 12a1.007 1.007 0 0 1-.302.715l-4.991 4.992a.997.997 0 0 1-1.414 0 .999.999 0 0 1 0-1.414L20.585 13H1a1 1 0 0 1-1-1V7a1 1 0 0 1 1-1Z"
                  fill="#222"
                  fill-rule="evenodd"
                />
              </svg>
              <a
                class="c8 fi-link fi-link--router c9"
                href="/"
              >
                Last will and testament
              </a>
            </span>
          </li>
        </ul>
      </li>
      <li
        class="c6 c7 fi-side-navigation-item fi-side-navigation-item--level-1"
      >
        <span
          class="c2 fi-side-navigation-item_content-wrapper"
        >
          <a
            class="c8 fi-link c11 fi-link fi-link--router c9 fi-link--external"
            href="https://suomi.fi"
            rel="noopener"
            target="_blank"
          >
            Taxation and public economy
            <span
              class="c2 c12 fi-visually-hidden"
            />
          </a>
        </span>
      </li>
      <li
        class="c6 c7 fi-side-navigation-item fi-side-navigation-item--level-1"
      >
        <span
          class="c2 fi-side-navigation-item_content-wrapper"
        >
          <button
            class="fi-link fi-link--router c9"
          >
            Consumer protection
          </button>
        </span>
      </li>
      <li
        aria-disabled="true"
        class="c6 c7 fi-side-navigation-item fi-side-navigation-item--level-1 fi-side-navigation-item--disabled"
      >
        <span
          class="c2 fi-side-navigation-item_content-wrapper"
        >
          <a
            class="c8 fi-link fi-link--router c9"
          >
            Finance
          </a>
        </span>
      </li>
    </ul>
  </nav>
</div>
`;
